datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -858,6 +850,13 @@ mod tests {
         }};
     }
 
+    macro_rules! test_coercion_like_rule {
```

Comment:
I wonder why this use a macro? It looks like this macro is only used in one place. Maybe it can be inlined?